c10/cuda/CUDACachingAllocator.h

```diff
@@ -50,73 +50,6 @@ namespace c10::cuda::CUDACachingAllocator {
 
 extern const size_t kLargeBuffer;
 
-struct Stat {
-  int64_t current = 0;
-  int64_t peak = 0;
-  int64_t allocated = 0;
-  int64_t freed = 0;
-};
-
-enum struct StatType : uint64_t {
-  AGGREGATE = 0,
-  SMALL_POOL = 1,
-  LARGE_POOL = 2,
-  NUM_TYPES = 3 // remember to update this whenever a new stat type is added
-};
-
-typedef std::array<Stat, static_cast<size_t>(StatType::NUM_TYPES)> StatArray;
-
-// Struct containing memory allocator summary statistics for a device.
-struct DeviceStats {
```

Comment:
This is the entry that some internal system looks for.
Can you re-expose the new version here to preserve the public API that was exposed from this header?

Reply:
I expose `c10::CachingDeviceAllocator::DeviceStats` as `c10::cuda::CUDACachingAllocator::DeviceStats` for BC. Could I capture your point?